test/functional/ex_cmds/mksession_spec.lua

```diff
@@ -0,0 +1,53 @@
+-- Test generation of session files
+
+local lfs = require('lfs')
+local helpers = require('test.functional.helpers')(after_each)
+
+local clear = helpers.clear
+local command = helpers.command
+local get_pathsep = helpers.get_pathsep
+local eq = helpers.eq
+local funcs = helpers.funcs
+
+
+describe('A session file', function()
+  -- The file name use for each session
+  local session_file = 'Xtest-functional-ex_cmds-mksession.vim'
+  -- Directory for testing tab-local working directories
+  local tab_dir = 'Xtest-functional-ex_cmds-mksession_spec.tab'
+
+  before_each(function()
+    clear()
+    lfs.mkdir(tab_dir)
+  end)
+
+  -- Each test should start with a clean session file
+  after_each(function()
+    os.remove(session_file)
+    lfs.rmdir(tab_dir)
+  end)
+
+  it('restores tab-local working directories', function()
+    local cwd_dir = funcs.getcwd()
+
+    -- There need to be some files being edited, so we will use dummy files
+    command('edit Xtest-functional-ex_cmds-mksession-tmpfile1')
```

Comment:
Better move to near other `Xtest-…` occurrences: in case somebody will want to move test file around this would be handy. Actually I would write the whole thing as

```lua
local file_prefix = 'Xtest-functional-ex_cmds-mksession'
local session_file = file_prefix .. '.vim'
-- …
local tab_dir = file_prefix .. '.d'
-- …
local tmpfile_base = file_prefix .. '-tmpfile'
```